### PR TITLE
chore(compiler): Use meaningful names in ast_helper

### LINKED ISSUE #1628: Compiler: Use meaningful module names in ast_helper
We currently use shorthands like `Imp` that are not clear as to what they represent.

### PR DESCRIPTION
**Pull request by ospencer**
Closes #1628

### CODE REVIEW

**Review by mortax — APPROVED** [review]
nice cleanup. 👍 for mnemonic names.

File: compiler/src/typed/typetexp.re [hunk 1]
```diff
@@ -13,7 +13,7 @@
 /*                                                                        */
 /**************************************************************************/
 
-/* typetexp.ml,v 1.34.4.9 2002/01/07 08:39:16 garrigue Exp */
+/* typetexp.ml,v 1.34.4.9 2002/01/07 08:39:16 garrigue Expression */
```
Comment (phated) [inline compiler/src/typed/typetexp.re:16]: Can we just drop this comment? Doesn't seem related to the license or anything.
```suggestion
```

**Review by phated — APPROVED** [review]
Just one comment, but I think this change will be great for first-time compiler contributors ❤️
